Source/FFmpegMediaSource.cpp

```diff
@@ -2397,31 +2413,69 @@ namespace winrt::FFmpegInteropX::implementation
             return -1;
         }
 
-        // Check beginning of file for BOM on first read
-        if (mss->streamByteOrderMark == ByteOrderMark::Unchecked)
+        // If we succeed but don't have any bytes, assume end of file
+        if (bytesRead == 0)
+        {
+            return AVERROR_EOF;  // Let FFmpeg know that we have reached eof
+        }
+
+        // Check encoding in case of external subtitle parser
+        if (!mss->streamEncodingChecked && mss->config->IsExternalSubtitleParser)
         {
-            if (bytesRead >= 4)
+            // read at least 1kb for first probing
+            bool isEof = false;
+            while (bytesRead < min(bufSize,1024))
             {
-                auto bom = ((uint32_t*)buf)[0];
-                if ((bom & 0x00FFFFFF) == 0x00BFBBEF)
+                ULONG read = 0;
+                hr = mss->fileStreamData->Read(buf + bytesRead, bufSize - bytesRead, &read);
+                if (FAILED(hr))
                 {
-                    mss->streamByteOrderMark = ByteOrderMark::UTF8;
+                    return -1;
                 }
-                else
+                else if (read == 0)
                 {
-                    mss->streamByteOrderMark = ByteOrderMark::Unknown;
+                    isEof = true;
+                    break;
                 }
+
+                bytesRead += read;
             }
-            else
+
```

Comment:
What's the purpose of the read-loop above? 
Before entering the if clause (!streamEncodingChecked), we've already read a non-zero amount of bytes into the buffer. Isn't that sufficient for the BOM check?

Reply:
I have seen a  web stream returning a single byte on first read call. But you are right, for BOM check, 4 bytes would be sufficient.